packages/angular/src/client/vcd.api.client.ts

```diff
@@ -242,6 +333,56 @@ export class VcdApiClient {
         );
     }
 
+    /**
+     * Creates an authenticated session for the specified credential data using cloud api endpoint.
+     *
+     * @param username the name of the user to authenticate
+     * @param tenant the organization the user belongs to
+     * @param password the password for the user
+     * @returns an authenticated session for the given credentials
+     */
+    public cloudApiLogin(username: string, tenant: string, password: string): Observable<Session> {
+        this._isCloudApiLogin = true;
+        const authString: string = btoa(`${username}@${tenant}:${password}`);
+        let url = `${this._baseUrl}/cloudapi/1.0.0/sessions`;
+        if (tenant.toLowerCase() === "system") {
+            url += "/provider";
+        }
+        return this.http.post<Session>(
+            url,
+            null,
+            {
+                observe: 'response',
+                headers: new HttpHeaders({ [HTTP_HEADERS.Authorization]: `Basic ${authString}`})
+            }
+        )
+        .pipe(
+            tap((response: HttpResponse<Session>) => {
+                // tslint:disable-next-line:max-line-length
+                const token = `${response.headers.get('x-vmware-vcloud-token-type')} ${response.headers.get('x-vmware-vcloud-access-token')}`;
+                // Call setCloudApiAuthentication since it not only sets the token but also updates the session, links etc
+                this.setCloudApiAuthentication(token).pipe(take(1)).subscribe(()=>{});
```

Comment:
You can use rxjs operators to subscribe to setCloudApiAuthentication when the user subscribes to cloudApiLogin, keep in mind that at the end you have to still return Observable<Session>.  By doing that we are going to avoid race potential conditions.